compiler/rustc_metadata/src/locator.rs

```diff
@@ -1208,10 +1219,25 @@ impl CrateError {
                 "plugin `{}` only found in rlib format, but must be available in dylib format",
                 crate_name,
             ),
-        };
+        }
+    }
 
-        err.emit();
+    crate fn report(self, sess: &Session, span: Span, missing_core: bool) {
+        self.build_diag(sess, span, missing_core).emit();
+    }
+
+    crate fn report_and_throw(self, sess: &Session, span: Span, missing_core: bool) -> ! {
```

Comment:
"throw" is a bad name for this, Rust doesn't have exceptions (and I think this is actually calling `process::exit` under the hood. I'd call it `report_fatal` instead.